cube_helper/cube_help.py

```diff
@@ -146,18 +227,38 @@ def add_categorical(cater_name, cubes, coord='time', season='djf',
         cubes: An iterable of Cubes, either a list of loaded Cubes
         or an iris CubeList.
     """
-    if isinstance(cubes, iris.cube.CubeList) or isinstance(cubes, list):
-        for cube in cubes:
-            _add_categorical(cater_name, cube,
-                             coord, season, seasons)
+    if isinstance(cater_name, list):
+        for categorical in cater_name:
+            if isinstance(cubes, iris.cube.CubeList) or isinstance(cubes, list):
+                for cube in cubes:
+                    _add_categorical(cube, categorical, coord, season, seasons)
+            else:
+                _add_categorical(cubes, categorical, coord, season, seasons)
         return cubes
-
     else:
-        _add_categorical(cater_name, cubes,
-                         coord, season, seasons)
-        return cubes
+        if isinstance(cubes, iris.cube.CubeList) or isinstance(cubes, list):
```

Comment:
I've made this change but removed it as it was failing tests, this is a weird one. Basically isinstance(cube, Iterable) will return true even with a regular cube, it will therefore try to iterate through it and raise a type error.